src/lib/Interfaces.ts

```diff
@@ -27,6 +27,9 @@ export interface IGoogleOAuth2Credential {
  */
 export interface IIdentity {
   uniqueID?: string | undefined;
+  displayName?: string | undefined;
+  verified?: boolean | undefined;
+  financialAccounts?: IFinancialAccount[] | undefined;
```

Comment:
@kim-formless Right now IUser contains an IIdentity object, and IFinancialAccount lives at the IUser level. I think we may want to add `financialAccounts` to IUser instead of IIdentity. That said, the dapp right now operates on the single primary financial account schema––so some associated changes would need to be made to support multiple accounts. Is there a concrete use case for this today, or can we get by with the primary financial account model given our CUJs?

Reply:
@un7c0rn hm, it seems redundant to have a single `financialAccount` map at the top level of `User` and also a `financialAccounts` nested one level down, but currently in the dApp we don't actually push any self-custodied wallet data to the DDN anywhere - we're just manually adding as a stopgap.`User.Identity.financialAccounts` operationally is our means of reconciling users like ourselves who have unique ID's but have uploaded using self-custodied wallets, so somewhat of a bridge between present and future state until users with fully managed wallets are able to upload. While not the cleanest, it does feel like a safe choice for now given that it doesn't change the single primary financial account schema and require updates to other areas of the code.

Reply:
@kim-formless If we do add a `financialAccounts` dict to IUser, we can mark `financialAccount` for deprecation, eg. keep both and then remove as a second step at the necessary time. That said, what's the need for storing multiple accounts with regard to the profile CUJ?

Reply:
@un7c0rn for the profile CUJ, storing multiple accounts was in response to storing the top level `self_custodied_evm_blockchain_address` key as another `financialAccount` dict instead. So in the case of my identity, I have both a fully managed and a self-custodied wallet, and when my profile renders, it fetches uploads and splits for both wallets. Here's its entry in the DB if it helps to clarify: https://console.cloud.google.com/firestore/databases/-default-/data/panel/verifiable_data_registry/0000000002?project=share-dev-345919

Reply:
@kim-formless Super helpful. My concern was that you'd have to do a pretty big backend refactor to move `financial_account` semantics to `financial_accounts`, but after looking at the code I now see that the backend is already framed around `financial_accounts` [1].

Thinking through frontend implications:

On the frontend, the `session` data type will serialize whatever exists in IUser [2] meaning if we add `financialAccounts` we should see `financialAccounts` in the local storage session dict.

During onboarding, although the backend stores a `financial_accounts` (plural) array, the dapp will currently _always_ set the primary session financial account to the 0 index account stored in the backend per [3] & [4]. The `sessionContext` does not have a setter for the plural `financialAccounts` key, instead if operates as if there is a single primary account in the session [5].

Based on PR https://github.com/formless-eng/share/pull/2040 it seems like for the profile CUJ, adding `displayName`, `verified` to the VDR schema, in conjunction with the existing `financialAccounts` key that is stored in the VDR, will unblock fetching profile info, and we actually don't need a client side `financialAccounts` key in the session yet (eg. we're still effectively operating the app as a single primary EVM address for now)––does that make sense?

[1] https://github.com/formless-eng/share/blob/4f4a09623409c3816e4499ad368699094e9d1929/backend/server_db.js#L1623

[2] https://github.com/formless-eng/share/blob/4f4a09623409c3816e4499ad368699094e9d1929/frontend/src/data_model/Session.ts#L24

[3] https://github.com/formless-eng/share/blob/4f4a09623409c3816e4499ad368699094e9d1929/frontend/src/components/onboard_view/OnboardView.tsx#L397

[4] https://github.com/formless-eng/share/blob/4f4a09623409c3816e4499ad368699094e9d1929/backend/methods/unique_id_get_managed_evm_account.ts#L46

[5] https://github.com/formless-eng/share/blob/4f4a09623409c3816e4499ad368699094e9d1929/frontend/src/components/onboard_view/OnboardView.tsx#L403